target/linux/ramips/dts/mt7620a_iptime_a1004ns.dts

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7620a_iptime.dtsi"
+
+/ {
+	compatible = "iptime,a1004ns", "ralink,mt7620a-soc";
+	model = "ipTIME A1004ns";
+
+	aliases {
+		led-boot = &led_cpu;
+		led-failsafe = &led_cpu;
+		led-running = &led_cpu;
+		led-upgrade = &led_cpu;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_cpu: cpu {
+			label = "a1004ns:blue:cpu";
+			gpios = <&gpio0 11 GPIO_ACTIVE_LOW>;
+		};
+
+		usb {
+			label = "a1004ns:blue:usb";
+			gpios = <&gpio1 15 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&ohci_port1>, <&ehci_port1>;
+			linux,default-trigger = "usbport";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+&firmware {
+	reg = <0x30000 0xfd0000>;
+};
+
+&state_default {
+	gpio {
+		ralink,group = "i2c", "uartf", "spi refclk";
+		ralink,function = "gpio";
+	};
+};
+
+&ethernet {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rgmii1_pins &mdio_pins>;
+
+	port@5 {
+		status = "okay";
+		mediatek,fixed-link = <1000 1 1 1>;
+		phy-mode = "rgmii";
+	};
+
+	mdio-bus {
+		status = "okay";
+
+		ethernet-phy@0 {
+			reg = <0>;
+			phy-mode = "rgmii";
+		};
+
+		ethernet-phy@1 {
+			reg = <1>;
+			phy-mode = "rgmii";
+		};
+
+		ethernet-phy@2 {
+			reg = <2>;
+			phy-mode = "rgmii";
+		};
+
+		ethernet-phy@3 {
+			reg = <3>;
+			phy-mode = "rgmii";
+		};
+
+		ethernet-phy@4 {
+			reg = <4>;
+			phy-mode = "rgmii";
+		};
+
+		ethernet-phy@1f {
+			reg = <0x1f>;
+			phy-mode = "rgmii";
+		};
```

Comment:
Do you actually need all these nodes? It seems that our eth driver will set up external mt7530 automatically with only the 0x1f node.
Could you test if it works without ethernet-phy@0~ethernet-phy@4 nodes?

Reply:
I tried removing phy0~phy4 and port 0 (wan port) stopped working. I needed (at least) phy0 to make it work.
I'll try to investigate this more.

Reply:
I tried removing phy0~phy4 and port 0 (wan port) stopped working. I needed (at least) phy0 to make it work.
I'll try to investigate this more.

Reply:
So this seems to work fine.
```
	mdio-bus {
		status = "okay";

		ethernet-phy@0 {
			reg = <0>;
		};

		ethernet-phy@1f {
			reg = <0x1f>;
			phy-mode = "rgmii";
		};
```
But I don't feel that's much different from the current state... I don't even understand why I need phy0 here.

Reply:
WN-AC1167GR had the same issue when I worked for supporting.

> It seems that our eth driver will set up external mt7530 automatically with only the 0x1f node.

I read the driver source and noticed it when that work, and I also deleted phy0 - phy4. But only port0 became unable to communicate. So, for the time being, I left all phys.